core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/archive/encoding/BinaryDecoder.java

```diff
@@ -432,10 +433,8 @@ public void checkBuffer(final int requiredLength) throws MALException
 
             // this either shifts the existing contents to the start of the old buffer, or copies it into the new buffer
             // NOTE: this is faster than System.arraycopy, as that performs argument type checks
-            for (int i = 0; i < existingContentRemaining; ++i)
-            {
-              destBuf[i] = this.buf[this.offset + i];
-            }
+              if (existingContentRemaining >= 0)
+                  System.arraycopy(this.buf, this.offset + 0, destBuf, 0, existingContentRemaining);
```

Comment:
The code comment note seems to imply that we should leave as it is.

Reply:
It seems so. I'll let the code as original was until i make some benchmark

Reply:
Changed